modules/extension/wmts/src/main/java/org/geotools/data/wmts/response/WMTSGetCapabilitiesResponse.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.data.wmts.response;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.commons.io.IOUtils;
+import org.geotools.data.ows.GetCapabilitiesResponse;
+import org.geotools.data.ows.HTTPResponse;
+import org.geotools.data.wmts.model.WMTSCapabilities;
+import org.geotools.ows.ServiceException;
+import org.geotools.wmts.WMTSConfiguration;
+import org.geotools.xml.Parser;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import net.opengis.wmts.v_1.CapabilitiesType;
+
+/**
+ * Provides a hook up to parse the capabilties document from inputstream.
+ *
+ * (Based on existing work by rgould for WMS service)
+ * @author ian
+ * @author Emanuele Tajariol (etj at geo-solutions dot it)
+ *
+ */
+public class WMTSGetCapabilitiesResponse extends GetCapabilitiesResponse {
+
+    public WMTSGetCapabilitiesResponse(HTTPResponse response) throws ServiceException, IOException {
+        this(response, null);
+    }
+
+    public WMTSGetCapabilitiesResponse(HTTPResponse response, Map<String, Object> hints) throws ServiceException, IOException {
+        super(response);
+
+        try {
+
+            Object object;
+            InputStream inputStream = null;
+            try {
+                inputStream = response.getResponseStream();
+                Parser parser = new Parser(new WMTSConfiguration());
```

Comment:
The parser is a throw away, but the configuration is shareable, please move it to a class field. (or at least, it should be, in GeoServer most configurations are Spring singletons)